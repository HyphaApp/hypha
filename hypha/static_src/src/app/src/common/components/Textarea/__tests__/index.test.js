import React from "react";
import { mount } from "enzyme";
import sinon from "sinon";
import TextArea from "../index";
import * as enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
enzyme.configure({ adapter: new Adapter() });


describe("Test textarea component", () => {
  const name = "test name";
  const label = "test label";
  const required = false;
  const choices = [["1", "one"], ["2", "two"]];
  const helperProps = {};
  const value = "1";
  const widget = {attrs : { cols : "a", rows : "c"}}
  const kwargs = { fields : [{"id": 1}, {"choices" : [1,2]}]}
  const onChange = jest.fn();


  const subject = mount(<TextArea
    name={name}
    label={label}
    required={required}
    choices={choices}
    value={value}
    helperProps={helperProps}
    widget={widget}
    kwargs={kwargs}
    onChange={onChange}
  />);

  it("Shoud render without issues", () => {
    expect(subject.length).toBe(1);
  });

  it("Should have searched classes and span element with passed text", () => {
    expect(subject.find('.form__group').length).toBe(2);
    expect(subject.find('.form__required').length).toBe(0);
    expect(subject.children().length).toEqual(1);
    expect(subject.find('span').first().text()).toBe(label);
    subject.find('textarea').props().onChange({currentTarget : { value : 1}})
    expect(onChange).toHaveBeenCalled();
    expect(onChange).toHaveBeenCalledWith(name, 1)
  });

  test('render a TextArea component', () => {
    expect(subject).toMatchSnapshot();
  });

});

describe("Test textarea component with required", () => {
  const name = "test name";
  const label = "test label";
  const required = true;
  const choices = [["1", "one"], ["2", "two"]];
  const helperProps = {};
  const value = "1";
  const widget = {attrs : { cols : "a", rows : "c"}}
  const kwargs = { fields : [{"id": 1}, {"choices" : [1,2]}]}
  const onChange = jest.fn();


  const subject = mount(<TextArea
    name={name}
    label={label}
    required={required}
    choices={choices}
    value={value}
    helperProps={helperProps}
    widget={widget}
    kwargs={kwargs}
    onChange={onChange}
  />);

  it("Shoud render without issues", () => {
    expect(subject.length).toBe(1);
  });

  it("Should have searched classes and span element with passed text", () => {
    expect(subject.find('.form__required').length).toBe(1);
  });

  test('render a TextArea component with required', () => {
    expect(subject).toMatchSnapshot();
  });

});
