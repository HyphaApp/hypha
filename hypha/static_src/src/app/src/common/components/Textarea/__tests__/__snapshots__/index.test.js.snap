// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test textarea component render a TextArea component 1`] = `
<Textarea
  choices={
    Array [
      Array [
        "1",
        "one",
      ],
      Array [
        "2",
        "two",
      ],
    ]
  }
  helperProps={Object {}}
  kwargs={
    Object {
      "fields": Array [
        Object {
          "id": 1,
        },
        Object {
          "choices": Array [
            1,
            2,
          ],
        },
      ],
    }
  }
  label="test label"
  name="test name"
  onChange={
    [MockFunction] {
      "calls": Array [
        Array [
          "test name",
          1,
        ],
      ],
    }
  }
  required={false}
  value="1"
  widget={
    Object {
      "attrs": Object {
        "cols": "a",
        "rows": "c",
      },
    }
  }
>
  <div
    className="form__group "
  >
    <div
      className="form__group  form__group--wrap"
      data-word-limit="1000"
    >
      <label
        className="form__question form__question--char_field textarea"
        htmlFor="test name"
      >
        <span>
          test label
        </span>
      </label>
      <HelperComponent />
      <div
        className="form__item"
      >
        <textarea
          cols="a"
          id="test name"
          onChange={[Function]}
          rows="c"
          value="1"
        />
      </div>
    </div>
  </div>
</Textarea>
`;

exports[`Test textarea component with required render a TextArea component with required 1`] = `
<Textarea
  choices={
    Array [
      Array [
        "1",
        "one",
      ],
      Array [
        "2",
        "two",
      ],
    ]
  }
  helperProps={Object {}}
  kwargs={
    Object {
      "fields": Array [
        Object {
          "id": 1,
        },
        Object {
          "choices": Array [
            1,
            2,
          ],
        },
      ],
    }
  }
  label="test label"
  name="test name"
  onChange={[MockFunction]}
  required={true}
  value="1"
  widget={
    Object {
      "attrs": Object {
        "cols": "a",
        "rows": "c",
      },
    }
  }
>
  <div
    className="form__group "
  >
    <div
      className="form__group  form__group--wrap"
      data-word-limit="1000"
    >
      <label
        className="form__question form__question--char_field textarea"
        htmlFor="test name"
      >
        <span>
          test label
        </span>
        <span
          className="form__required"
        >
           *
        </span>
      </label>
      <HelperComponent />
      <div
        className="form__item"
      >
        <textarea
          cols="a"
          id="test name"
          onChange={[Function]}
          rows="c"
          value="1"
        />
      </div>
    </div>
  </div>
</Textarea>
`;
